### PR TITLE
nmbayes: updates for _log compatibility

### LINKED ISSUE #40: make _log() helpers work with nmbayes submodels
Follow-up to gh-30

`config_log`, `run_log` and `summary_log` get tripped up by the nested sampling models.  The fix will likely involve adjusting bbr's `find_models` to ignore these submodels.

```
$ git rev-parse HEAD
589c984ed7120fff202c89c46c782d410dc84740
$ git grep FIXME tests
tests/testthat/test-config-log.R:  skip("FIXME: submodels confuse config_log()")
tests/testthat/test-workflow-nmbayes.R:  skip("FIXME: submodels confuse run_log()")
tests/testthat/test-workflow-nmbayes.R:  skip("FIXME: submodels confuse summary_log()")
```

### PR DESCRIPTION
**Pull request by kyleam**
In-flight changes on bbr's side make `run_log`, `config_log`, and `summary_log` compatible with nmbayes models:

 * https://github.com/metrumresearchgroup/bbr/pull/643
 * https://github.com/metrumresearchgroup/bbr/pull/644
 * https://github.com/metrumresearchgroup/bbr/pull/645

This series implements the bbr.bayes-side of the `config_log` fix and drops the skips for the `_log` tests.

Closes #40

---

- [x] Replace tip commit (Drone configuration) once the changes land in bbr's main

~- [ ] Bump bbr version constraint?  My current thinking is that we should leave it as is and assess things after changes for gh-36 come in.~  Tracked at https://github.com/metrumresearchgroup/bbr.bayes/issues/124

### CODE REVIEW

**Comment by kyleam** [comment]
This is ready for a review (alongside the bbr PRs), but merging should wait until everything lands on bbr's side.

File: tests/testthat/test-workflow-nmbayes.R
Comment (seth127) [inline tests/testthat/test-workflow-nmbayes.R:85]: Is the point of adding `.recurse = TRUE` here just to make sure that the nested models no longer show up? I'm assuming yes, but correct me if I'm missing something.

**Review by seth127 — CHANGES_REQUESTED** [review]
This is mostly looking great, but a few things to address:

## `summary_log` fix and regression test

First, address my `summary_log` comment from the `bbr` PR: https://github.com/metrumresearchgroup/bbr/pull/645#discussion_r1472279335 . Whatever the outcome of that is, we should probably have something in `test-summary-log.R` to check it's what we expect. (Currently that file only tests `stan_summary_log()` and doesn't call `bbr::summary_log()` anywhere.)

## `run_log()` regression test

In my testing, `test-run-log.R` passed both before and after the associated `bbr` changes. This is because we only test for the [presence of an `nmbayes` model](https://github.com/metrumresearchgroup/bbr.bayes/blob/6fe5d0f00444e0320d4c147ecb6992528d2a50ff/tests/testthat/test-run-log.R#L8-L10), and we _don't_ test for the absence of the submodels. 

Before these changes, this test was passing on a table that looked like this:

Screenshot 2024-01-30 at 9 41 16 PM

Consider adding a regression test of sorts, to check that the submodels are not getting picked up.

File: tests/testthat/test-workflow-nmbayes.R
Comment (kyleam) [reply tests/testthat/test-workflow-nmbayes.R:85]: Using `.recurse = TRUE` does demonstrate that the nested models don't show up, but that's not the main reason for it.  This `run_log` call uses "." as the directory, and that looks like this:

```
extdata/
model/nonmem/bayes/
```

So we need the `.recurse = TRUE` for this to get any models at all.

Here's my explanation for "how did that work before?" in the message of 4b37b8d:

> [...] update the run_log() call to pass `.recurse = TRUE` to account
> for the change in default (bbr's afa6bff6).

That lines up date-wise:

 * [ed33da2](https://github.com/metrumresearchgroup/bbr.bayes/commit/ed33da2415df1356b6ddd2101ec67c498e46cc8c) (tests: record nmbayes failures for *_log functions, 2022-11-16)

 * [afa6bff6](https://github.com/metrumresearchgroup/bbr/commit/afa6bff65f8511125c01f5176c1d60e630b2a57b) (refactor: changing all _log() functions to default to .recurse = FALSE [...], 2022-11-28)

**Comment by kyleam** [comment]
> In my testing, `test-run-log.R` passed both before and after the associated `bbr` changes.

Thanks for testing that.

> Consider adding a regression test of sorts, to check that the submodels are not getting picked up.

Two spots already serve to verify that submodels aren't included (one nmbayes-specific):

 * [bbr tests](https://github.com/metrumresearchgroup/bbr/blob/b2ca32bf1a26189b2b0891f5c5069187c7c7b4b4/tests/testthat/test-run-log.R#L181-L262) that came with metrumresearchgroup/bbr#643

 * the [nmbayes run_log tests](https://github.com/metrumresearchgroup/bbr.bayes/blob/4b37b8dac02b3cf823286425509d00611a5cfdcc/tests/testthat/test-workflow-nmbayes.R#L84-L90) in `test-workflow-nmbayes.R`

But it's a fair point that the dedicated run_log test should do so too.  Will update.

**Comment by kyleam** [comment]
> we should probably have something in `test-summary-log.R` to check it's what we expect.  (Currently that file only tests `stan_summary_log()` and doesn't call `bbr::summary_log()` anywhere.)

Right, that file is focused on testing the `R/summary-log.R` file from this repo (so the tailored `stan_*` summary functions). 
 Would you fine with doing that check in `test-workflow-stan.R` (which already calls `bbr::summary_log()`), or would you prefer it in `test-summary-log`?

**Comment by kyleam** [comment]
I said:

> But it's a fair point that the dedicated run_log test should do so too. Will update.

Done with 9e2d7e7.

**Comment by seth127** [comment]
Thanks for the updates @kyleam 

> Right, that file is focused on testing the R/summary-log.R file from this repo (so the tailored stan_* summary functions).
Would you fine with doing that check in test-workflow-stan.R (which already calls bbr::summary_log()), or would you prefer it in test-summary-log?

Thanks for the explanation. I think it's fine just having it in `test-workflow-stan.R`, and it's also [in `test-workflow-nmbayes.R`](https://github.com/metrumresearchgroup/bbr.bayes/blob/9e2d7e7e84885e79653e52a95fae96073c7adb1f/tests/testthat/test-workflow-nmbayes.R#L92-L99).

**Review by seth127 — DISMISSED** [review]
I think this looks good. The one thing I'm still a little stuck on is this

> address my `summary_log` comment from the `bbr` PR: https://github.com/metrumresearchgroup/bbr/pull/645#discussion_r1472279335 . Whatever the outcome of that is, we should probably have something in `test-summary-log.R` to check it's what we expect.

Essentially, once we make whatever change we decide on, should we have a test that it shows up the way we expect in `summary_log()` (and where should that test live).

I can see an argument that the thing we should be testing is that `model_summary()` returns what we expect, which will then propogate to `model_summaries()` and then on to `summary_log()`. But that's enough moving pieces that I do kinda want some level of "integration test" for it. I think that can wait until whatever PR addresses the change to `model_summary()` though (if we add it at all).

**Comment by kyleam** [comment]
> The one thing I'm still a little stuck on is this

Once those `model_summary` changes land, I agree that there should be a test on this side showing we get the expected value.

What do you think about doing the `model_summary` change as part of this PR, assuming we decide to go that direction?  (related to my comment [here](https://github.com/metrumresearchgroup/bbr/pull/645#discussion_r1473260521))

Scope-wise, it seems to fit well enough given the motivation for it is fixing `summary_log`.

**Comment by seth127** [comment]
As noted in https://github.com/metrumresearchgroup/bbr/pull/645#discussion_r1473479118 let's go ahead with the `model_summary()` changes, and I'm fine with them happening in this PR if that is @kyleam 's preference.

**Comment by kyleam** [comment]
Updates:

 * renamed `summary-log.R` and `test-summary-log.R` to `stan-summary-log.R` and `test-stan-summary-log.R` to 1) make it clearer that these are about `stan_summary_log()` and to make way for the `bbr::summary_log()` tests (4edbe36)

 * `model_summary` methods now error; `test-summary-log.R` has test of `summary_log` result for directory containing `nonmem`, `nmbayes`, and `stan` model types  (a0f4e99)

**Comment by kyleam** [comment]
The test failures are unrelated (gh-125).  Assuming that's triggered by rate limiting, I'm going to let things sit for a bit and then restart, but @seth127 this is ready for you to take another look.

**Review by seth127 — APPROVED** [review]
These changes all look good to me. For posterity, the bulk of the changes relevant to [this `summary_log()`/`model_summary()` discussion](https://github.com/metrumresearchgroup/bbr/pull/645#discussion_r1472279335) are in 327df7379df965be74f9ecdf81e74cab57bba002. Thanks @kyleam .

**Comment by kyleam** [comment]
@seth127 Thanks for taking another look (and for catching the issue with putting draws in the summary_log output).

**Comment by kyleam** [comment]
Pushed planned update to make Drone pull from bbr's `main` instead of the (now merged) PR branch.  Will merge after CI completes.

**Pull request merged**
